src/solvers/flattening/arrays.cpp

```diff
@@ -123,10 +123,7 @@ void arrayst::collect_arrays(const exprt &a)
     collect_arrays(a.op0());
 
     // make sure this shows as an application
-    index_exprt index_expr;
-    index_expr.type()=array_type.subtype();
-    index_expr.array()=a.op0();
-    index_expr.index()=a.op1();
+    index_exprt index_expr(a.op0(), a.op1(), array_type.subtype());
```

Comment:
Nit picking: the third operand is unnecessary. Also: would it be possible to use `to_with_expr` above so that the rather anonymous `a.op0()` and `a.op1()` become more readable?